### ISSUE TITLE
If change is made to improve mtime logic, remote buffer may be briefly considered to have a filesystem conflict

### ISSUE THREAD

**Issue by mgsloan** [zed-industries/zed#21034]
[`collab tests::integration_tests::test_buffer_conflict_after_save`](https://github.com/zed-industries/zed/blob/0559e1f348507caac8d9b5d2a6484ce398e163a4/crates/collab/src/tests/integration_tests.rs#L3493) is fails when uses of [`bad_is_greater_than`](https://github.com/zed-industries/zed/blob/0559e1f348507caac8d9b5d2a6484ce398e163a4/crates/fs/src/fs.rs#L222) are removed in favor of using mtime equality check instead.

[On file save](https://github.com/zed-industries/zed/blob/6c470748ac481db9e31503d7349ead286f9886b3/crates/project/src/buffer_store.rs#L455), downstream clients are first notified of the file change and then notified of the buffer change.  This passes a `>` check but does not pass a `==` check, and so with the improved mtime logic this is briefly considered a conflict.

**Comment by notpeter** [comment]
@mgsloan Is this still valid?

**Comment by mgsloan** [comment]
@notpeter Yes, still valid, but the description was quite stale, I've updated it to reflect the current state of things.  Not sure when I'll get around to addressing this, so unassigning.
